lib/ts/utils.ts

```diff
@@ -179,6 +179,12 @@ export async function validateForm(
  * getCurrentNormalisedUrlPath
  */
 export function getCurrentNormalisedUrlPath(): NormalisedURLPath {
+    try {
+        WindowHandlerReference.getReferenceOrThrow().windowHandler.getWindowUnsafe();
+    } catch (_) {
+        return new NormalisedURLPath("/");
```

Comment:
I don't think this is correct. For example, this'll make `canHandleRoute` return false everywhere if the `basePath` is not `/` and true everywhere if it is. Not sure how bad this is but it can cause some unexpected things.
This is only used in a few places, where we could make it overrideable by adding an optional param that'll skip calling this.